### PR TITLE
Porting of REC 3.3 ICT Equipment

### PR DESCRIPTION
**Pull request by PeteHart**
Uses the REC namespace not to interfere with future BRICK Equipment subclasses.

The class hierarchy is similar to that of the REC 3.3 ICT Equipment class. In particular equipments for retrofitted sensors are hence available. As is various networking equipments.

Some properties have been changed. Notably
- Height in RUs and number of ports properties have been moved to the ICT Equipment class and hence inhered by all its subclasses.
- Also on the ICT Equipment class a standard property has been added should reference to a specific standard be desired (e.g. IEEE 802.11).
- POE type on ethernet switch ports has been enumerated.
- For wireless access points type has been changed to generation, and the possible generations are enumerated.

No relationships have been ported. Use of the existing REC 4 relationships are encouraged.

### CODE REVIEW

File: Source/DTDLv2/RealEstateCore/Asset/Equipment/ICTEquipment/ICTEquipment.json [hunk 1]
```diff
@@ -0,0 +1,41 @@
+{
+  "@id": "dtmi:org:w3id:rec:ICTEquipment;1",
+  "@type": "Interface",
+  "description": {
+    "en": "Equipment and devices that are part of a building's ICT infrastructure."
+  },
+  "displayName": {
+    "en": "ICTEquipment"
```
Comment (hammar) [inline Source/DTDLv2/RealEstateCore/Asset/Equipment/ICTEquipment/ICTEquipment.json:8]: "ICT Equipment" with a space?

File: Source/DTDLv2/RealEstateCore/Asset/Equipment/ICTEquipment/ICTEquipment.json [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+      "name": "numberOfPorts",
+      "displayName": {
+        "en": "Number of Ports"
+      },
+      "schema": "integer",
+      "writable": true
+    },
+    {
+      "@type": "Property",
+      "name": "standard",
```
Comment (hammar) [inline Source/DTDLv2/RealEstateCore/Asset/Equipment/ICTEquipment/ICTEquipment.json:31]: Can we add a description to this Property?

File: Source/DTDLv2/RealEstateCore/Asset/Equipment/ICTEquipment/DataNetworkEquipment/EthernetSwitchPort.json [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+            "displayName": "802.3bt Type 4, 4PPoE or PoE++",
+            "enumValue": "Type4"
+          }
+        ]
+      },
+      "writable": true
+    },
+    {
+      "@type": "Property",
+      "name": "portSpeed",
```
Comment (hammar) [inline Source/DTDLv2/RealEstateCore/Asset/Equipment/ICTEquipment/DataNetworkEquipment/EthernetSwitchPort.json:47]: Should this be a string or an enum field? Remember that we discussed it but don't recall the specifics. Are there too many possibilities for an enum?

File: Source/DTDLv2/RealEstateCore/Asset/Equipment/ICTEquipment/ICTHardware/ICTHardware.json [hunk 4]
```diff
@@ -0,0 +1,12 @@
+{
+  "@id": "dtmi:org:w3id:rec:ICTHardware;1",
```
Comment (hammar) [inline Source/DTDLv2/RealEstateCore/Asset/Equipment/ICTEquipment/ICTHardware/ICTHardware.json:2]: If we've removed the "Server" subclass, what other thing would be "ICT Hardware" but not "ICT Equipment"? Might it be worthwhile to keep Server and drop this class?

**Review by hammar — COMMENTED** [review]
Thanks Peter, this looks nice! I've left some comments on things that are a little unclear to me, but this might be things we discussed already and I forgot to note..? 

See also the ongoing work in Brick that I've linked from this PR discussion. I don't know how fast they'll progress, but ideally we won't be duplicating functionalities for too long..

**Comment by epaulson** [comment]
Rather than saying 'EthernetRouter', maybe you'd rather say 'NetworkRouter'?

I may be in the minority on the Brick side too, but I would rather see Controllers and the gateway under the 'DataNetwork' part of the tree. I think someday a 'PLC' would wind up in there too.

File: Source/DTDLv2/RealEstateCore/Asset/Equipment/ICTEquipment/ICTEquipment.json (hunk 2 above)
Comment (PeteHart) [reply Source/DTDLv2/RealEstateCore/Asset/Equipment/ICTEquipment/ICTEquipment.json:31]: Agree, not self-explanatory, will add

File: Source/DTDLv2/RealEstateCore/Asset/Equipment/ICTEquipment/DataNetworkEquipment/EthernetSwitchPort.json (hunk 3 above)
Comment (PeteHart) [reply Source/DTDLv2/RealEstateCore/Asset/Equipment/ICTEquipment/DataNetworkEquipment/EthernetSwitchPort.json:47]: We did discuss an enum. Then when implementing I found this https://en.wikipedia.org/wiki/List_of_interface_bit_rates (see the LAN section) and decided against it.

Not strongly against it though. We can of course stick to the traditional Ethernet port speeds only, like 10BASE-X, 100BASE-X etc. or only enumerate the bandwith without mentioning (or implicitly referring to) the standard, eg 10Mbit/s, 100Mbit/s...

Can we consider briefly at upcoming TC?

File: Source/DTDLv2/RealEstateCore/Asset/Equipment/ICTEquipment/ICTHardware/ICTHardware.json (hunk 4 above)
Comment (PeteHart) [reply Source/DTDLv2/RealEstateCore/Asset/Equipment/ICTEquipment/ICTHardware/ICTHardware.json:2]: I think I forgot adding the Server. Will fix.

The I suppose there can be equipments such a power supplies / UPS that someone would want to add!?

**Comment by PeteHart** [comment]
As to the point on 'EthernetRouter' -> 'NetworkRouter' which I agree to, I wonder if it would be better to actually only use Router and Switch, i.e. no prefixed Ethernet or Network? For example, A switch may be a Layer 3 Switch which is not only Ethernet switching.

File: Source/DTDLv2/RealEstateCore/Asset/Equipment/ICTEquipment/ICTEquipment.json (hunk 1 above)
Comment (PeteHart) [reply Source/DTDLv2/RealEstateCore/Asset/Equipment/ICTEquipment/ICTEquipment.json]: Fixed with new commit.

File: Source/DTDLv2/RealEstateCore/Asset/Equipment/ICTEquipment/ICTEquipment.json (hunk 2 above)
Comment (PeteHart) [reply Source/DTDLv2/RealEstateCore/Asset/Equipment/ICTEquipment/ICTEquipment.json:31]: Fixed with new commit.

File: Source/DTDLv2/RealEstateCore/Asset/Equipment/ICTEquipment/ICTHardware/ICTHardware.json (hunk 4 above)
Comment (PeteHart) [reply Source/DTDLv2/RealEstateCore/Asset/Equipment/ICTEquipment/ICTHardware/ICTHardware.json:2]: Fixed with new commit.

File: Source/DTDLv2/RealEstateCore/Asset/Equipment/ICTEquipment/DataNetworkEquipment/EthernetSwitchPort.json (hunk 3 above)
Comment (hammar) [reply Source/DTDLv2/RealEstateCore/Asset/Equipment/ICTEquipment/DataNetworkEquipment/EthernetSwitchPort.json:47]: Suggestion: let's update this to use quantitative type extension from DTDLv3 preview.
E.g., the portspeed is a Property and a DataRate, with schema integer or float, and unit mebibitPerSecond or mebibytePerSecond

Comment (hammar) [reply Source/DTDLv2/RealEstateCore/Asset/Equipment/ICTEquipment/DataNetworkEquipment/EthernetSwitchPort.json:47]: https://github.com/Azure/opendigitaltwins-dtdl/blob/master/DTDL/Docs/en-US/DTDL.quantitativeTypes.v1.md

File: Source/DTDLv2/RealEstateCore/Asset/Equipment/ICTEquipment/DataNetworkEquipment/EthernetSwitchPort.json [hunk 5]
```diff
@@ -0,0 +1,57 @@
+{
+  "@id": "dtmi:org:w3id:rec:EthernetSwitchPort;1",
```
Comment (hammar) [inline Source/DTDLv2/RealEstateCore/Asset/Equipment/ICTEquipment/DataNetworkEquipment/EthernetSwitchPort.json:2]: Rename to EthernetPort

File: Source/DTDLv2/RealEstateCore/Asset/Equipment/ICTEquipment/DataNetworkEquipment/EthernetRouter.json [hunk 6]
```diff
@@ -0,0 +1,12 @@
+{
+  "@id": "dtmi:org:w3id:rec:EthernetRouter;1",
```
Comment (hammar) [inline Source/DTDLv2/RealEstateCore/Asset/Equipment/ICTEquipment/DataNetworkEquipment/EthernetRouter.json:2]: Rename to NetworkRouter

**Review by hammar — CHANGES_REQUESTED** [review]

**Comment by hammar** [comment]
Note for reference, so we don't miss closing later: this PR closes #211

File: Source/DTDLv2/RealEstateCore/Asset/Equipment/ICTEquipment/DataNetworkEquipment/EthernetSwitchPort.json (hunk 3 above)
Comment (PeteHart) [reply Source/DTDLv2/RealEstateCore/Asset/Equipment/ICTEquipment/DataNetworkEquipment/EthernetSwitchPort.json]: Changed with recent commit.

File: Source/DTDLv2/RealEstateCore/Asset/Equipment/ICTEquipment/DataNetworkEquipment/EthernetRouter.json (hunk 6 above)
Comment (PeteHart) [reply Source/DTDLv2/RealEstateCore/Asset/Equipment/ICTEquipment/DataNetworkEquipment/EthernetRouter.json]: Fixed in recent commit.

File: Source/DTDLv2/RealEstateCore/Asset/Equipment/ICTEquipment/DataNetworkEquipment/EthernetSwitchPort.json (hunk 5 above)
Comment (PeteHart) [reply Source/DTDLv2/RealEstateCore/Asset/Equipment/ICTEquipment/DataNetworkEquipment/EthernetSwitchPort.json]: Changed in recent commit.

**Pull request merged**